feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -739,6 +780,26 @@ class BillingWrapper(
         }
     }
 
+    private fun BillingClient.queryProductDetailsAsyncEnsuringOneResponse(
+        params: QueryProductDetailsParams,
+        listener: ProductDetailsResponseListener
+    ) {
+        var hasResponded = false
+        queryProductDetailsAsync(params) { billingResult, productDetailsList ->
+            synchronized(this@BillingWrapper) {
```

Comment:
Hmm I guess `queryProductDetailsAsync` can call the callback multiple times? 😞

Reply:
Not sure 🤷‍♀️

Reply:
context https://github.com/RevenueCat/purchases-android/pull/369
although tbh we should add a comment here providing that context so we don't have to wonder or dig it up